compiler/rustc_next_trait_solver/src/solve/assembly/mod.rs

```diff
@@ -301,16 +302,14 @@ impl<'tcx> EvalCtxt<'_, InferCtxt<'tcx>> {
 
         let mut candidates = vec![];
 
-        self.assemble_non_blanket_impl_candidates(goal, &mut candidates);
+        self.assemble_impl_candidates(goal, &mut candidates);
```

Comment:
This consolidates `assemble_{blanket,non_blanket}_impl_candidates`. See the definition of `for_each_relevant_impl` that I added to `Interner`.